### ISSUE TITLE
GAL & ATF programming

### ISSUE THREAD

**Issue by LobAnovich** [chris-jh/MegaCD-Connect-BD-Remake#3]
Greetings !  I received 2 + 2 GAL22V10B chips from Ali from different sellers.  The chips are erased, programmed, but do not pass verification unless the “verification after recording” checkbox is unchecked.  Is it possible that this is a “lock bit” action and for the same reason it is impossible to remove a dump from the original chip or are these still fake chips?  Do I need to leave the “lock bit” and “encrypt chip” checkboxes before recording?  Programmer TL866II-Plus.  Xgpro 12.66 program

The situation with programming the ATF22V10C is slightly different.  The chips are erased, programmed, and verified at the end of the programming process, but if you try to re-verify separately, the verification is immediately loaded with exactly the same error as in the case of GAL during the programming process.  This gave me the idea that this situation arose due to the operation of the “lock bit”.  Or are all the chips faulty?

**Comment by chris-jh** [comment]
Have you tried them yet?

I can't remember getting verification errors and i didnt mess with any lock bits or encryption.

years ago go i did use the gui version, but Mac is my main os and i use the open source minipro command line version now and i never specified them options.

If they dont work in the board, then i would assume fake or faulty.

unless it is the minipro you have ? an original or a clone ? i bought mine direct from xgecu

**Comment by LobAnovich** [comment]
> Have you tried them yet?
> 
> I can't remember getting verification errors and i didnt mess with any lock bits or encryption.
> 
> years ago go i did use the gui version, but Mac is my main os and i use the open source minipro command line version now and i never specified them options.
> 
> If they dont work in the board, then i would assume fake or faulty.
> 
> unless it is the minipro you have ? an original or a clone ? i bought mine direct from xgecu

I have the original Xgecu TL866II-Plus programmer, which I have been using for many years.  I just identified a completely faulty batch of logic 74374 using this programmer, which I bought for assembling a connection board.  Lots 7410 and 74245 are fully functional.  Unfortunately, in addition to completely faulty logic, I have two addons that still need additional repairs and I’m not sure that in their current state I will be able to check the operation of the new connection board, given the programming problems

![IMG_20240430_175328](https://github.com/Board-Folk/MegaCD-Connect-BD-Remake/assets/113496444/28907f62-8d63-4db1-944e-f1c7b3f2f5e0)

Now,with two working connection boards, on one of the addons, I have graphics artifacts and animations of the BIOS loading screen, as well as the addon freezing when I try to launch the game disc.  The data on the game disk is detected by the addon, but when you try to launch it, the addon freezes.  CDDA playback fine

**Comment by LobAnovich** [comment]
As I wrote earlier, the Chinese sent ATF22V10CQZ-20PC chips instead of ATF22V10B-15PC.  After several checks, I was able to write, count, and check a randomly selected chip, like ATF22V10C, several times, with a write voltage of 12 Volts and with the “lockbit” checkbox unchecked before writing.

A little later I’ll try to check the remaining 4 units of these chips

**Comment by LobAnovich** [comment]
![IMG_20240505_234358](https://github.com/Board-Folk/MegaCD-Connect-BD-Remake/assets/113496444/47fe40b1-851c-4e9a-9f09-e6d50f6cc996)

![IMG_20240505_234540](https://github.com/Board-Folk/MegaCD-Connect-BD-Remake/assets/113496444/dc095e65-af06-4f33-b320-5051530cf03d)


I restored my Sega mega cd addon and checked the adapter board clone.  I haven't tested it for a long time, so far it works fine

**Comment by LobAnovich** [comment]
Also, all different ATF22V10B chips from different sellers, which were normally programmed and verified, do not work normally in sonic cd. The picture in the game itself, crumbles into squares, and then the game freezes. Only one version of GAL22V10B chips, from a large number that I bought from different sellers and which were normally flashed, has no problems in operation.

**Comment by chris-jh** [comment]
What logic chips you are using ? i had issues with IC4 thinking it was something with the GAL Code. But it was that chip, which tested good in logic tests. But it was wrong/fake from the seller. Swapped that chip from ones i bought from mouser all good. it delayed this project a year. Shakes fists at fake chip sellers....

**Comment by chris-jh** [comment]
Basically the Logic Chips needs to be the types specified on the BOM IC4 needs to be an ACT

**Comment by LobAnovich** [comment]
ATF22V10B-15PC. ATF22V10CQZ-20PU
These are the chips that are flashed and verified normally. The project works with these chips, but the gameplay of the sonic cd does not work correctly.

With the original IC4 I have the same problems as with the replacement one

**Comment by LobAnovich** [comment]
Although... I reflashed the original IC4 with a multi-regional BIOS and there were much fewer artifacts with the ATF22V10B-15PC. I even managed to pass the first level. The second level started without any artifacts at all. It is quite possible that my ATF22V10B-15PC are actually relabeled from slower ATF22V10B-20,25 or even slower ones. As an experiment, I will try to find a faster chip for IC4 (-70 or -10). Finding a real working GAL22 or ATF22 is a very big problem.

**Comment by LobAnovich** [comment]
> What logic chips you are using ? i had issues with IC4 thinking it was something with the GAL Code. But it was that chip, which tested good in logic tests. But it was wrong/fake from the seller. Swapped that chip from ones i bought from mouser all good. it delayed this project a year. Shakes fists at fake chip sellers....


What IC4 are you talking about? The logic chip on the connector board? I assumed that you were writing about the IC4 on the main board. And I replaced the BIOS chip with the original one reflashed with a multibios))) Although this also gave a result. A very positive result. 
As for the IC4 on the connector board, I have 74HC10 installed, but there is also 74ACT10. Both versions pass tests on the programmer. I can also try replacing HC with ACT

**Comment by chris-jh** [comment]
IC4 on the connect-bd board. I know the original peel board has all HC, and i can't remember why i put ACT in the bom now. But i was getting lots of corruption and things, when this IC was not correct.

Other thing, did you try  with and without them caps back on the back of your board also ?

Sega, being sega used to like adjusting timings by adding caps it seems.

Was wondering if the boards you are trying do need them or not

**Comment by chris-jh** [comment]
I bought the ATF22V10C15PU from mouser (556-AF22V10C15PU) they have plenty of stock and not EOL as far as i know. These are what i used.

Someone also pointed out to me that the ATF22V10 have a problem. The last fuse enables a "power down" mode on pin 4, and if it is erased it defaults to on. If you don't have a programmer that knows this you get odd behaviour when power down mode goes on.

**Comment by LobAnovich** [comment]
![IMG_20241215_014244](https://github.com/user-attachments/assets/e655f593-ea09-49d1-b463-8bdc8c36a33d)

I installed ceramic capacitors for regulation, with the values ​​as on the original board that I had.

**Comment by chris-jh** [comment]
wonder what happens with them removed between the resistor arrays.

Like i said, Sega when it sees issues with certain machines, they like to tweak data lines manually by adding random caps 😂

**Comment by LobAnovich** [comment]
> wonder what happens with them removed between the resistor arrays.
> 
> Like i said, Sega when it sees issues with certain machines, they like to tweak data lines manually by adding random caps 😂

I agree, but they do the adjustment manually not on each individual copy or even a separate batch, but on the entire batch, which was assembled from certain, identical components. That is, if, let's say, they started using any chip from another manufacturer and problems arise during testing, then they modify all copies in the same way.

And yes, I removed the delay-regulating capacitors, both one by one and two at once, and this does not affect the problem in any way.

**Comment by LobAnovich** [comment]
I replaced 74HC with 74ACT and it got a little worse. I can still complete levels (the game doesn't freeze), but now the picture completely falls apart into squares, as it was today, before replacing the BIOS chip, only then the game also froze at the beginning of the first level. Although, if I use GAL instead of ATF, then there are no problems

**Comment by LobAnovich** [comment]
> I bought the ATF22V10C15PU from mouser (556-AF22V10C15PU) they have plenty of stock and not EOL as far as i know. These are what i used.
> 
> Someone also pointed out to me that the ATF22V10 have a problem. The last fuse enables a "power down" mode on pin 4, and if it is erased it defaults to on. If you don't have a programmer that knows this you get odd behaviour when power down mode goes on.

Unfortunately, I only have access to chips from China, and even then they are mostly used and far from the full range.

The first time I flashed ATF.., using XGecu TL866II pro. The second time I flashed ATF... using XGecu T48. I don't know if these programmers know about the "power down fuse" or not. How should this "feature" manifest itself when the connection board with the ATF chip works...?

**Comment by chris-jh** [comment]
Did you get this sorted ?

**Comment by LobAnovich** [comment]
> Did you get this sorted ?

Yes, I bought some new GAL22V10B-10LP and some new AT27C1024-70PU (write-once). Now, with any test, there are no problems in operation. Testing with 74ACT10, instead of 74HC10 showed deterioration in operation

**Comment by LobAnovich** [comment]
[MegaCD-Connect BD Remake_final.zip](https://github.com/user-attachments/files/19755773/MegaCD-Connect.BD.Remake_final.zip)

I also asked a friend to remake the source files for the production of the connection board, taking into account the change in the power lines and the installation of the capacitor. I ordered the production of boards according to these files, connection boards, but I have not yet checked the correctness.

**Comment by Sham-San** [comment]
> [MegaCD-Connect BD Remake_final.zip](https://github.com/user-attachments/files/19755773/MegaCD-Connect.BD.Remake_final.zip)
> 
> I also asked a friend to remake the source files for the production of the connection board, taking into account the change in the power lines and the installation of the capacitor. I ordered the production of boards according to these files, connection boards, but I have not yet checked the correctness.

Hi, is there any news. If I order a new version of the board, what are the best chips to use from Aliexpress and what capacitors are needed

**Comment by LobAnovich** [comment]
> > [MegaCD-Connect BD Remake_final.zip](https://github.com/user-attachments/files/19755773/MegaCD-Connect.BD.Remake_final.zip)
> > I also asked a friend to remake the source files for the production of the connection board, taking into account the change in the power lines and the installation of the capacitor. I ordered the production of boards according to these files, connection boards, but I have not yet checked the correctness.
> 
> Hi, is there any news. If I order a new version of the board, what are the best chips to use from Aliexpress and what capacitors are needed

This variant is completely identical to the original project and is intended for the same chips + corrected power lines, in accordance with the power lines of the original board + space for a capacitor with a capacity not less than the nominal value installed on the original adapter board. More is possible.

I ordered the chips through a Chinese intermediary. Chips I got from Aliexpress were faulty.

**Comment by Sham-San** [comment]
Is it possible to use some other Programmable Logic instead of GAL22V10B-15 / ATF22V10B-15

**Comment by LobAnovich** [comment]
> Is it possible to use some other Programmable Logic instead of GAL22V10B-15 / ATF22V10B-15

I am not the author of this project and have no information about the possibility of using other controllers. I tested only with GAL22 and ATF22

As I wrote earlier, I tested the connection boards for this project on several mega cd copies. On some, in very good condition, with stock project components, there were no problems, on some there was a problem with Sonic CD. On some there were problems with the add-on (mega cd). The operation of the connection board for this project depends on the condition of the add-on motherboard.

That's why I started experimenting. I found that replacing 74ACT10 chips with 74HC10 slightly improves the situation. Replacing the ATF22 controller with GAL22V10B-10 also greatly improves stability. Using faster chips for the BIOS (AT27C1024-70PU (write-once)) also has a positive effect on stability. That's why I assembled a test unit on these components, which showed maximum efficiency and stability. If problems remained, they were problems on the mega cd motherboard side.

I do not exclude the presence of problems in the operation of the connection board on the standard components of the project, perhaps re-marked controllers or rewritable chips for the BIOS with a long access time were used. The ID numbers of these chips were checked before programming

**Comment by LobAnovich** [comment]
And although the following actions took me a lot of time,as an experiment, I even completely checked one of the Mega CD motherboards, did preventive maintenance by shunting more than 250 vias, removing all the chips and checking absolutely all the lines, and then replacing all the memory modules with new ones, the fastest ones I could get. Of course, I did a FRAM mod on the save module. This did not change the situation in the operation of the connection board for this project with stock components.

**Comment by Sham-San** [comment]
> [MegaCD-Connect BD Remake_final.zip](https://github.com/user-attachments/files/19755773/MegaCD-Connect.BD.Remake_final.zip)
> 
> I also asked a friend to remake the source files for the production of the connection board, taking into account the change in the power lines and the installation of the capacitor. I ordered the production of boards according to these files, connection boards, but I have not yet checked the correctness.

What is the capacitance of C7?
C1-C6 is 0.1 μF?

**Comment by LobAnovich** [comment]
> > [MegaCD-Connect BD Remake_final.zip](https://github.com/user-attachments/files/19755773/MegaCD-Connect.BD.Remake_final.zip)
> > I also asked a friend to remake the source files for the production of the connection board, taking into account the change in the power lines and the installation of the capacitor. I ordered the production of boards according to these files, connection boards, but I have not yet checked the correctness.
> 
> What is the capacitance of C7? C1-C6 is 0.1 μF?

C1-C6 (0.1 µF).  C7 (47-100 uF)

**Comment by Sham-San** [comment]
Hi. I used Gerber boards version 2.2 from this thread, GAL22V10B-10LP, 74HC374, 74HC245, 74HC10, 0.1 µF capacitors, and a 47 µF electrolytic capacitor. All games from any region work perfectly on the Japanese version of the console with the stock BIOS.

**Comment by DeVllllL** [comment]
> Hi. I used Gerber boards version 2.2 from this thread, GAL22V10B-10LP, 74HC374, 74HC245, 74HC10, 0.1 µF capacitors, and a 47 µF electrolytic capacitor. All games from any region work perfectly on the Japanese version of the console with the stock BIOS.

Hello. Good day.

I also had the PCB manufactured. Where did you buy the GAL22V10B-10LP ICs?

**Comment by Sham-San** [comment]
> > Hi. I used Gerber boards version 2.2 from this thread, GAL22V10B-10LP, 74HC374, 74HC245, 74HC10, 0.1 µF capacitors, and a 47 µF electrolytic capacitor. All games from any region work perfectly on the Japanese version of the console with the stock BIOS.
> 
> Hello. Good day.
> 
> I also had the PCB manufactured. Where did you buy the GAL22V10B-10LP ICs?

AliExpress

**Comment by LobAnovich** [comment]
This seller (lihengic.88@vip.163.com) has 69 stores on AliExpress. Contact him and confirm whether he sells and ships to your country. I've already purchased nine GAL22 chips from him, and the chips have always been new, working, and not relabeled. I'm still waiting for an order from him with three of these chips.

**Comment by DeVllllL** [comment]
> This seller ([lihengic.88@vip.163.com](mailto:lihengic.88@vip.163.com)) has 69 stores on AliExpress. Contact him and confirm whether he sells and ships to your country. I've already purchased nine GAL22 chips from him, and the chips have always been new, working, and not relabeled. I'm still waiting for an order from him with three of these chips.

I also bought the ATF22V10C-15PU from Digikey, just like you.

But I don't know if it will work, so I'm asking where to buy the GAL22V10B-10LP from a reliable seller on AliExpress.

**Comment by DeVllllL** [comment]
> > > Hi. I used Gerber boards version 2.2 from this thread, GAL22V10B-10LP, 74HC374, 74HC245, 74HC10, 0.1 µF capacitors, and a 47 µF electrolytic capacitor. All games from any region work perfectly on the Japanese version of the console with the stock BIOS.
> > 
> > 
> > Hello. Good day.
> > I also had the PCB manufactured. Where did you buy the GAL22V10B-10LP ICs?
> 
> AliExpress

Can you share the link to the ad where I bought it, please?

**Comment by LobAnovich** [comment]
Through this aggregator I bought three units of GAL22V10B-10LP three times. Each time he creates the lot I need in the quantity I need, through his different stores on AliExpress.

**Comment by DeVllllL** [comment]
> Through this aggregator I bought three units of GAL22V10B-10LP three times. Each time he creates the lot I need in the quantity I need, through his different stores on AliExpress.

Okay. I googled the email address you gave me and the AliExpress store Liheng Technology (HK) Limited came up. Is that one of them? I sent them a message asking about the chips. We'll see what they say.

**Comment by DeVllllL** [comment]
A few days ago, the PCBs arrived. I've already soldered all the components. I also programmed the IC I bought from Digikey, the ATF22V10C-15PU. And the Mega CD now powers on!! However, it's currently frozen on the main BIOS screen, and the "Access" and "Ready" LEDs aren't lighting up. Do you think it's because I'm not using the GAL22V10B-10LP?

Anyway, I ran a RAM test on the Mega CD, and it's showing errors on all four RAM modules.

I programmed the ATF22V10C-15PU using the file from the repository: MegaCD-Connect-BD-Remake/Logic/GAL22V10/GAL22V10.jed

**Comment by chris-jh** [comment]
What logic chips are being used, i had fake 74ACT10's that caused me issues

**Comment by chris-jh** [comment]
It is important to stay to the same series of logic chip as i specified ACT and HC.

The ACT one is important to get correct

**Comment by DeVllllL** [comment]
> It is important to stay to the same series of logic chip as i specified ACT and HC.
> 
> The ACT one is important to get correct

I bought them all from Digikey, exactly all the components listed on the repository's main page, using Digikey's codes. The only thing I couldn't find were the GAL22V10B-15 / ATF22V10B-15, and the closest thing I had was the ATF22V10C-15PU-ND

It could be the CD drive; it's the only thing I haven't recapped. I did recap the other PCBs.

**Comment by chris-jh** [comment]
MegaCD's Main PCB's are also notorious for corroded via's, even after a recap. Also why Cosam reproduced the MegaCD PCB.

**Comment by DeVllllL** [comment]
> MegaCD's Main PCB's are also notorious for corroded via's, even after a recap. Also why Cosam reproduced the MegaCD PCB.

Yes, I know the issue. I'm going to thoroughly inspect my PCB. Initially, when I replaced the capacitors, I did a quick check and didn't see any leaking capacitors or corroded traces. Besides, I cleaned the entire PCB with IPA.

The first thing I'm going to do is buy the GAL22V10B-10LP chips, and while I wait for them to arrive, I'll replace the capacitors in the CD drive.

**Comment by LobAnovich** [comment]
Three brand new chips arrived. Nothing was written to their memory. I wrote and verified all three. The main thing is to disable the "lock bit" option when writing, otherwise the re-verification of the written information against the original will fail.

![Image](https://github.com/user-attachments/assets/239c98ba-11a1-4840-b88f-b208457d4067)


And one more thing. When I experimented with different chips on a working add-on, I had no problems launching the add-on itself. There were some issues with running games. But I must admit, I tested all the logic chips with a programmer before soldering them in. So, absolutely all the chips used in the build were tested and found to be working.

**Comment by LobAnovich** [comment]
Now I have an mega cd on its way with a similar problem and it is definitely not a problem with the connection board.

![Image](https://github.com/user-attachments/assets/08ec4fab-0f38-4fec-816a-9d67a94d18d8)

**Comment by DeVllllL** [comment]
> Three brand new chips arrived. Nothing was written to their memory. I wrote and verified all three. The main thing is to disable the "lock bit" option when writing, otherwise the re-verification of the written information against the original will fail.
> 
> ![Image](https://github.com/user-attachments/assets/239c98ba-11a1-4840-b88f-b208457d4067)
> 
> And one more thing. When I experimented with different chips on a working add-on, I had no problems launching the add-on itself. There were some issues with running games. But I must admit, I tested all the logic chips with a programmer before soldering them in. So, absolutely all the chips used in the build were tested and found to be working.

I did exactly the same as you. I programmed the chip without checking the bit lock.

I also did the same with the logic chips, testing them one by one with the T48 programmer; they all passed.

**Comment by DeVllllL** [comment]
> Now I have an mega cd on its way with a similar problem and it is definitely not a problem with the connection board.
> 
> ![Image](https://github.com/user-attachments/assets/08ec4fab-0f38-4fec-816a-9d67a94d18d8)

Exactly, that's the same error I'm getting.

It gets stuck on the home screen, but it freezes.

**Comment by scp-electrocution** [comment]
Hi everyone, I'm trying to repair my Mega CD, but unfortunately I can't compile the software for my ATF22V10C-10PU. I don't think the file already compiled for the GAL chip is compatible. Could someone help me compile the correct file? Thanks for your help.
I programmed with the already compiled file and it seems to work fine, I'll do some tests to see if everything is ok.

**Comment by chris-jh** [comment]
Yes use the pre-compiled version. There were some differences with the GALasm in the latest version that compiled different. I have now made a new folder for this version of GALasm, which also has the pre-compiled version in it too.

**Comment by chris-jh** [comment]
> Now I have an mega cd on its way with a similar problem and it is definitely not a problem with the connection board.
> 
> ![Image](https://github.com/user-attachments/assets/08ec4fab-0f38-4fec-816a-9d67a94d18d8)

That is usually when it can't talk to the CD Drive. like it is unplugged.

**Comment by LobAnovich** [comment]
> That is usually when it can't talk to the CD Drive. like it is unplugged.

Usually this is the case, but here the case is more complicated )))

![Image](https://github.com/user-attachments/assets/7423d7b4-305a-4ba6-9190-65066adea5db)


Based on the annotation for the new firmware, does it make sense to replace the previously installed one ?

**Comment by chris-jh** [comment]
No point they both compile to the same jed file map.

I tested both on my ATF chip for a few hours each running sonic cd mainly and also tried time gal and final fight cd.

Infact it was on a megacd i had that was not working not powering up.

I just change the 2 caps near the regulator for now, brought it back 😂

The whole machine needs new caps very grotty.

It is also one with an original peel chip (still works too).



Might be time for a new board on that one you have

**Comment by LobAnovich** [comment]
> No point they both compile to the same jed file map.

The firmware for version 2.1, when verified, differs from the firmware for version 2.3, but the firmware for version 2.1 itself has not been removed from the repository. Which firmware do you recommend using now?

**Comment by chris-jh** [comment]
As i said, either Jed file that i have compiled already. Does not matter.

The folder marked as fixed, is if you want to compile yourself using the latest GALasm in the gitrepo i link to.

**Comment by chris-jh** [comment]
> Version 2.3 Changes:-
> 
> Updates to Kicad v9, Changed A26/B26 on the CD Expansion to point to 9v, and not 5v. Made no difference with the older version, but made it now match the original PCB.
> 
> Added new Logic version, because GALasm was updated and was giving a different compiled version that does not work. Found what i needed to change to make it work.
> 
> **You can use either Pre-Compiled Jed Files, for the GAL22v10 / ATF22V10**
